src/webgpu/util/constants.ts

```diff
@@ -466,17 +476,17 @@ export const kValue = {
       min: reinterpretU16AsF16(kBit.f16.positive.min),
       max: reinterpretU16AsF16(kBit.f16.positive.max),
       zero: reinterpretU16AsF16(kBit.f16.positive.zero),
-      first_f64_not_castable: reinterpretU16AsF16(kBit.f16.positive.max) / 2 + 2 ** 16, // mid point of 2**16 and largest f16
-      last_f64_castable: reinterpretU64AsF64(
+      first_webidl_not_castable: reinterpretU16AsF16(kBit.f16.positive.max) / 2 + 2 ** 16, // mid point of 2**16 and largest f16
```

Comment:
As Kai pointed out there seems to be a typo here. I revisited and I think it should be `reinterpretU16AsF16(kBit.f16.positive.max) / 2 + 2 ** 15, // mid point of 2**16 and largest f16`

`2**128` and `2**16` should be the infinity of f32 and f16

f32: exponent bit = 11111111<sub>2</sub> (is infinity in IEEE754) -> 2<sup>255-127</sup> -> `2**128`
f16: exponent bit = 11111<sub>2</sub> (is infinity in IEEE754) -> 2<sup>31-15</sup> -> `2**16`

(largest_f16 + infinity_16) / 2 = reinterpretU16AsF16(kBit.f16.positive.max) / 2 + 2 ** 15

Reply:
Hmmm... apparently I didn't commit a change on my branch before pushing. I will do that now.

I have actually removed the f16 forms of this, since from my quick reading of the WebIDL spec, https://webidl.spec.whatwg.org/#es-float, half float is not defined/the conversion is not specified at all.
Since nothing in the repo depends on it, it makes sense to me to remove it entirely.

Reply:
What I'm not sure is:

`2**128` is treated as a special value explicitly mentioned in [WebIDL](https://webidl.spec.whatwg.org/#es-float), when converting number/f64 to f32

But number/f64 to f16 conversion according to WebGPU/WGSL [spec](https://gpuweb.github.io/gpuweb/#conversions-to-wgsl) is first casting to wgslF32 then do f16(wgslF32). So taking the symmetry special value of f32 to fit f16 (`2**128` to `2**16`) doesn't make sense here. Maybe this is also the reason you didn't put the same comment here like you did for the f32 ones?

Kai: Did we have discussion about this when I add this code? I vaguely remember we did, and we found the case is over-complicated and we cannot fully test what we want. I think we may want to remove the `f16` ones and its usages to avoid confusion?

Reply:
> I have actually removed the f16 forms of this

Yeah that was what I want to type at the same time. Removing LGTM.

You also need to remove its usages in `compute_pipeline.spec.ts` and `overrides.spec.ts`
(search with`/f16\..*first_f64_not_castable/`)